services/QuillLMS/client/app/bundles/Staff/components/comprehension/changeLog/changeLog.tsx

```diff
@@ -0,0 +1,253 @@
+import { DropdownInput, Spinner, } from '../../../../Shared/index';
+
+interface ChangeLogProps {
+
+}
+
+const ChangeLog = ({ history, match }) => {
+  const { params } = match;
+  const { activityId, } = params;
+  const SESSION_INDEX=1;
+  const initialStartDateString = window.sessionStorage.getItem(`${SESSION_INDEX}startDate`) || '';
+  const initialEndDateString = window.sessionStorage.getItem(`${SESSION_INDEX}endDate`) || '';
+  const initialStartDate = initialStartDateString ? new Date(initialStartDateString) : null;
+  const initialEndDate = initialEndDateString ? new Date(initialEndDateString) : null;
+
+  const [searchInput, setSearchInput] = React.useState<string>('');
+  const [prompt, setPrompt] = React.useState<string>('all');
+  const [rule, setRule] = React.useState<string>('all');
+  const [startDate, onStartDateChange] = React.useState<Date>(initialStartDate);
+  const [endDate, onEndDateChange] = React.useState<Date>(initialEndDate);
+
+  const { data: changeLogData, status: status } = useQuery({
+    queryKey: [`activity-${activityId}`, activityId],
+    queryFn: fetchChangeLog
+  });
+
+  const { data: activityData } = useQuery({
+    queryKey: [``, activityId],
+    queryFn: fetchActivity
+  });
+
+  function handleSearch(e) {
+    setSearchInput(e.target.value)
+  }
+
+  function handlePromptChange(e) {
+    setPrompt(e.target.value)
+  }
+
+  function handleRuleChange(e) {
+    setRule(e.target.value)
+  }
+
+  const promptDropdown = (
+    <div id="prompt-dropdown">
+      <p className="control" >
+        <span className="select">
+          <select defaultValue='all' onChange={handlePromptChange}>
+            <option value="all">All Prompts</option>
+            <option value="because">because</option>
+            <option value="but">but</option>
+            <option value="so">so</option>
+          </select>
+        </span>
+      </p>
+    </div>
+  )
+
+  const formattedRows = changeLogData && changeLogData.changeLogs && changeLogData.changeLogs.map(log => {
+    const {
+      action,
+      changed_record_id,
+      updated_local_time,
+      previous_value,
+      new_value,
+      record_type_display_name,
+      user,
+      explanation,
+      conjunction,
+      name,
+      changed_attribute
+    } = log;
+
+    const changedRecord = `${record_type_display_name} - ${changed_record_id}`
+    const actionLink = explanation && JSON.parse(explanation).url
+    const prompt = explanation && JSON.parse(explanation).conjunction
+
+    return {
+      action: action,
+      changedRecord: changedRecord,
+      previousValue: previous_value,
+      newValue: new_value,
+      author: user,
+      dateTime: updated_local_time,
+      actionLink: actionLink,
+      prompt: prompt,
+      conjunction: conjunction,
+      name: name,
+      changedAttribute: changed_attribute
+    }
+  })
+
+  const filteredRows = formattedRows && formattedRows.filter(value => {
```

Comment:
It's not strictly necessary but I think it would be a lot cleaner to save the value of each `.filter` call in a function (ex: `filteredBySearchInput`, `filteredByConjunction` etc) or even to make each of them their own functions, kind of like we do in this file: https://github.com/empirical-org/Empirical-Core/blob/develop/services/QuillLMS/client/app/bundles/Teacher/components/assignment_flow/create_unit/custom_activity_pack/index.tsx#L234. If you think about debugging one of these functions, for instance, you'd going to have to rewrite a bunch of code rather than just printing the value of the specific filter that you think is getting missed to see what's going wrong.